### PR TITLE
(#18301): ignoring scss resources that lives in the same directory as the main scss file

### LINKED ISSUE #18301: SASS not working with multiple independent SCSS files. Causes hanging on front-end
> Reproducible on 5.2.8 demo site.

### Describe the bug

When creating multiple independent SCSS files that are referenced/compiled separately using built in compiler:
- The front-end hangs on the SASS files
- The .dotsass referenced files result in an HTTP 504 error.
- Page doesn't completely load correctly due to missing CSS files.

Screenshot of demo site after reproducing.

image

### Steps to reproduce the behavior:
1. Create a new test SCSS file with the following code, nothing fancy or complicated:

```
.row.no-gutter {
	margin-left: 0;
	margin-right: 0;

	div[class*='col-'] {
		padding-left: 0;
		padding-right: 0;
	}
}
```

2. Upload file to `/application/themes/travel/css`
2.5 Change the file extension to `.scss` at fields: `File Name` and `Title`
3. Edit  `/application/themes/travel/html_head.vtl` to reference new file

```
<link rel="stylesheet" href="${dotTheme.path}css/test.dotsass">
```

4. Save/Publish `html_head.vtl`
5. Everything should still be working at this point, including the new SCSS file.
6. Open `/application/themes/travel/css/styles.scss` for editing, lock the file.
7. Do not change contents of file, just Save/Publish.
8. Try to load `https://demo.dotcms.com/store/`, page will hang. The `.dotsass` referenced files will timeout with a 504 error. Once this happens the page will display, but without any styling.

### Expected behavior

Each `.dotsass` referenced file compiles independently of each other and everything is peachy king.

### Additional context

I have run into this issue on multiple clients' sites. If we want to use the built-in SASS compiler, we're stuck with one master SCSS file. This is usually not a problem, but there are instances where having independent SCSS files with specific targets in mind.

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Since they will have a shot with the `CssPreProcessServlet`. This way we will avoid recursively process each other as they were included in the compiler's `allImportedURI` attribute.

### CODE REVIEW

**Comment by wezell** [comment]
Question for you @vicozizou if you have 2 scss files in the top directory, I see you are ignoring the other one in there.  But do they both work, e.g. you have two files `/application/themes/css/testcss1.scss` and `/application/themes/css/testcss2.scss`. 
 Can you call either of them and they work?
`/application/themes/css/testcss1.dotsass`
`/application/themes/css/testcss2.dotsass`

**Review by freddyucv — APPROVED** [review]

**Comment by victoralfaro-dotcms** [comment]
@wezell 
I added both `test1.dotsasss` and `test2.dotsass` files to same location as `styles.dotsass` is.
image
Got the following:
image

File: dotCMS/src/main/java/com/dotcms/csspreproc/DotCSSCompiler.java [hunk 1]
```diff
@@ -234,6 +237,10 @@ protected File createCompileDir(Host currentHost, String uri, boolean live)
     return compDir;
   }
 
+  private boolean areSiblings(final String uri, final String otherUri) {
+    return uri.substring(0, uri.lastIndexOf("/")).equals(otherUri.substring(0, otherUri.lastIndexOf("/")));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/csspreproc/DotCSSCompiler.java:241]: not sure if this method should be promoted to some Util
Also use StringPool.SLASH

Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/csspreproc/DotCSSCompiler.java:241]: It totally can.

**Review by erickgonzalez — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**